src/crystal/system/unix/process.cr

```diff
@@ -194,7 +215,6 @@ struct Crystal::System::Process
     argv << Pointer(UInt8).null
 
     LibC.execvp(command, argv)
-    raise RuntimeError.from_errno
```

Comment:
Why's this removed without replacement?

Reply:
Because now the call from `Crystal::System::Process.spawn` assumes that if the call returns, it's because there was some error and it proceeds to send the `errno` through the pipe.

Reply:
Oh, wait... that breaks `Process.exec`, right? I'll check

Reply:
You're right. It should be fixed now. Also I added the `NoReturn` return type to `Process.exec` and another spec 😉